app/models/spree/product_decorator.rb

```diff
@@ -5,7 +5,11 @@ def variant_images
     nonuniq_variant_images.reorder(:position).distinct
   end
 
+  def gallery
+    variant_images || images || [Spree::Image.new]
```

Comment:
Does this need to return an instance of the new `ProductGallery` class?  I believe this needs to return something that has access to the `images` method.

Reply:
Good call, thanks for pointing that out. I'll change it up.